containers/daap-python-base/src/var/task/versioning.py

```diff
@@ -141,6 +175,61 @@ def metadata_update_type(data_product_metadata) -> UpdateType:
         return UpdateType.MinorUpdate
 
 
+def schema_update_type(
+    data_product_schema: DataProductSchema,
+) -> tuple[UpdateType, dict]:
+    """
+    Figure out whether changes to the input data represent a major or minor schema update
+    and return the changes as a dict, e.g.
+        {table_name: {columns:{...}, non_column_fields: {...}}}
+    """
+    if not data_product_schema.exists or not data_product_schema.valid:
+        return UpdateType.NotAllowed, {}
+
+    changed_fields = data_product_schema.changed_fields()
+
+    if "columns" in changed_fields:
+        column_changes = data_product_schema.detect_column_differences_in_new_version()
+
+        if any([column_changes["removed_columns"], column_changes["types_changed"]]):
+            update_type = UpdateType.MajorUpdate
+        elif any(
+            [column_changes["added_columns"], column_changes["descriptions_changed"]]
+        ):
+            update_type = UpdateType.MinorUpdate
+        elif not any(
+            [
+                column_changes["removed_columns"],
+                column_changes["types_changed"],
+                column_changes["added_columns"],
+                column_changes["descriptions_changed"],
+            ]
+        ):
+            update_type = UpdateType.Unchanged
+
+    if update_type == UpdateType.Unchanged:
+        if not changed_fields:
+            update_type = UpdateType.Unchanged
+        if not changed_fields.difference(MINOR_UPDATE_SCHEMA_FIELDS):
+            update_type = UpdateType.MinorUpdate
+
```

Comment:
this should be equivalent:

```suggestion
    if update_type == UpdateType.Unchanged and changed_fields and not changed_fields.difference(MINOR_UPDATE_SCHEMA_FIELDS):
        update_type = UpdateType.MinorUpdate
```

not sure I understand the edge case this is handling though. Is there anything that can change other than `columns`?

Reply:
currently only `tableDescription` although this is almost certain to increase. I'm going to look at this bit again